monitor/uidmonitor/processor.go

```diff
@@ -0,0 +1,346 @@
+package uidmonitor
+
+import (
+	"errors"
+	"fmt"
+	"regexp"
+	"strconv"
+	"strings"
+	"sync"
+
+	"go.uber.org/zap"
+
+	"github.com/aporeto-inc/trireme/cache"
+	"github.com/aporeto-inc/trireme/collector"
+	"github.com/aporeto-inc/trireme/monitor"
+	"github.com/aporeto-inc/trireme/monitor/contextstore"
+	"github.com/aporeto-inc/trireme/monitor/linuxmonitor/cgnetcls"
+	"github.com/aporeto-inc/trireme/monitor/rpcmonitor"
+	"github.com/aporeto-inc/trireme/policy"
+)
+
+// UIDProcessor captures all the monitor processor information for a UIDLoginPU
+// It implements the MonitorProcessor interface of the rpc monitor
+type UIDProcessor struct {
+	collector         collector.EventCollector
+	puHandler         monitor.ProcessingUnitsHandler
+	metadataExtractor rpcmonitor.RPCMetadataExtractor
+	netcls            cgnetcls.Cgroupnetcls
+	contextStore      contextstore.ContextStore
+	regStart          *regexp.Regexp
+	regStop           *regexp.Regexp
+	storePath         string
+	putoPidMap        *cache.Cache
+	pidToPU           *cache.Cache
+	sync.Mutex
+}
+
+const (
+	triremeBaseCgroup = "/trireme"
+)
+
+//puToPidEntry -- represents an entry to puToPidMap
+type puToPidEntry struct {
+	pidlist            map[string]bool
+	Info               *policy.PURuntime
```

Comment:
why this field is public? having public field in private struct is never a good idea